### PR TITLE
M3-1396 refactor linodes

### PR DESCRIPTION
**Pull request by Jskobos**
Refactor `services/linodes`, including:

* add comments
* standardize signatures
* add schemas, types, and interfaces where applicable
* remove unused methods
* refactor manager-ized methods into generics

### CODE REVIEW

File: src/features/linodes/LinodesDetail/LinodeBackup/LinodeBackup.tsx [hunk 1]
```diff
@@ -285,6 +286,18 @@ class LinodeBackup extends React.Component<CombinedProps, State> {
...
     const { linodeID } = this.props;
     const { settingsForm } = this.state;
+    /* The api will set a default day of week if none is provided, but if a day is
+    * provided without a window, it will return an error.
+    */
+    if (settingsForm.day && settingsForm.window) {
+      this.setState({
+        settingsForm: {
+          ...settingsForm,
+          errors: [{'reason': 'If a day of the week is provided, you must specify a time.'}]
```
Comment (martinmckenna) [inline src/features/linodes/LinodesDetail/LinodeBackup/LinodeBackup.tsx]: This error appears when you specify a time but not a day as well, so in that case, the error message doesn't make much sense

**Comment by martinmckenna** [comment]
Sorry to be a pain here - Now no error appears when you choose a backup day, but no time

File: src/services/linodes/linode.schema.ts [hunk 2]
```diff
@@ -1,14 +1,185 @@
...
+  stackscript_id: number().notRequired(),
+  backup_id: number().notRequired(),
+  swap_size: number().notRequired(),
+  image: string().nullable(true),
+  root_pass: root_pass.notRequired(),
+  authorized_keys: array().of(string()).notRequired(),
+  backups_enabled: boolean().notRequired(),
+  stackscript_data,
+  booted: boolean().notRequired(),
+  label: string().notRequired()
```
Comment (patthiel) [inline src/services/linodes/linode.schema.ts]: this `notRequired()` doesn't seem to be working as expected here. I can't seem to create a Linode without entering a label now

**Comment by patthiel** [comment]
When rebuilding, the power status indicator updates to only contain the chevron
screen shot 2018-10-26 at 11 05 32 am

**Comment by patthiel** [comment]
The progress indicator no longer displays when rebuilding a linode on linode detail

File: src/services/linodes/linode.schema.ts (hunk 2 above)
Comment (Jskobos) [reply src/services/linodes/linode.schema.ts]: Fixed this one, though not happy about the fix (explicitly passing null to the API methods when label hasn't been filled out). Ones below I wasn't able to reproduce.

**Pull request merged**